### ISSUE TITLE
Is anyone using with Okta Identity Engine?

### ISSUE THREAD

**Issue by btsteve** [Versent/saml2aws#1000]
Is anyone currently using saml2aws with Okta Identity Engine?
 We are looking to upgrade and want to check and see if other users were using it successfully.

**Comment by pnancarrow** [comment]
I tried and did not get very far. Okta told us that we needed to switch to their tool: https://github.com/okta/okta-aws-cli or move to AWS SSO.

**Comment by mikex99** [comment]
You can use it with OIE, but you have to switch to the Browser mfa method.

**Comment by mannesrikanth** [comment]
Saml2aws isn't working with OKTA Identity Engine, seems it has to be implemented.

**Comment by mikex99** [comment]
> Saml2aws isn't working with OKTA Identity Engine, seems it has to be implemented.

It does work with OIE, but you can't use the "Okta" method.  That method relies on API calls that no longer work with OIE.  You have to use the Browser MFA method with OIE.

**Comment by pnancarrow** [comment]
nice @mikex99 ! this works for me

**Comment by mannesrikanth** [comment]
Thank you @mikex99, seems a lot of OS dependency to download the drivers but finally I'm able to make it work. Thank you again.

**Comment by riedlem** [comment]
I just tested it last week, and again just now. It's worked for me with the "Okta" method (using okta verify) for our preview tenant that got upgraded to OIE

**Comment by mannesrikanth** [comment]
@riedlem  I just tested with "Okta" IDP on our OIE tenant and it didn't work, the authentication call went against the /authn endpoint in the command prompt and got 401 even with valid credentials. In OKTA Identity Engine, the device authorization and MFA (Based on sign on policies) should be performed over the browser in OIE.

**Comment by prema-loom** [comment]
@riedlem believe it works for you on upgraded preview tenant because Okta moves all apps to an auth policy that only requires password. Moving AWS to any other app auth policy needing 1 or 2 factors makes saml2aws error out with auth 401 error, if Okta is used as provider. It works if browser provider is used

**Comment by riedlem** [comment]
But wouldn't the "only requires password" policy not have the saml2aws prompt me for an okta verify passcode?

We just upgraded our okta prod environment to OIE at the end of July and so far things have been working like the preview environment did.

I still select okta as the provider, and it has me login with my password, and then the 6 digit token from my verify app. It then connects me and I can see that I'm given AWS keys with an expiration date, and can run AWS cli commands as before. I also haven't heard of any issues from our cloud team saying things aren't working anymore

**Comment by prema-loom** [comment]
@riedlem do you have Global Session Policy to require Password+MFA & also is AWS and other apps under "Classic Migrated Policy" - Policy created during OIE upgrade and applied to all applications that were protected by the default app-sign on policy in Classic.

**Comment by riedlem** [comment]
Yes, we have a Global Session Policy that requires MFA and prompts users every 12hrs. 
No, the AWS app isn’t listed under the “Classic Migrated Policy”, though we do have other apps that are under it.
